lib/msal-node/src/client/ManagedIdentityApplication.ts

```diff
@@ -35,6 +36,14 @@ import {
     DEFAULT_AUTHORITY_FOR_MANAGED_IDENTITY,
     ManagedIdentitySourceNames,
 } from "../utils/Constants.js";
+import { ManagedIdentityId } from "../config/ManagedIdentityId.js";
+import { HashUtils } from "../crypto/HashUtils.js";
+
+const SOURCES_THAT_SUPPORT_TOKEN_REVOCATION: Array<ManagedIdentitySourceNames> =
```

Comment:
Great that you have a dedicated list of sources that handle revocation. If more services adopt it, it should be trivial to add them to SOURCES_THAT_SUPPORT_TOKEN_REVOCATION.

Reply:
It won't be that easy, would it? Each MSIv1 provider is different. We will need to know their API version (which may or may not change, and if they do, other aspects of the protocol will likely also change).

(I am not necessarily against the current implementation, as long as it is functionally correct. I am just calling out the need to still be careful when more services adopt token revocation.)

Reply:
> It won't be that easy, would it? Each MSIv1 provider is different. We will need to know their API version (which may or may not change, and if they do, other aspects of the protocol will likely also change).

Agree

Reply:
Ray, it will actually be extremely trivial. Look at my PR for App Service token revocation [here](https://github.com/AzureAD/microsoft-authentication-library-for-js/pull/7772).